lib/rest/routes/api-endpoints.js

```diff
@@ -0,0 +1,30 @@
+const express = require('express');
+
+module.exports = function ({config}) {
+  let router = express.Router();
+  router.put('/:name', function (req, res, next) {
+    let isUpdate;
+    config.update(json => {
+      isUpdate = req.params.name in json.apiEndpoints;
+      json.apiEndpoints[req.params.name] = req.body;
```

Comment:
I think you guys already discussed this - so bear with me if I am repetitive - but this might be harmful as fundamentally I can put everything I want without having any sort of validation.

Reply:
I agree as for general idea, why it is not in this PR 

I cannot do it for pipeline policies, because it depends on introducing schema for policies in plugins. 

Here the only thing that is required is 'host' field. but for `paths` it is quite complex and I would like to move it after approach for pipelines validation is settled. (+ some intentionally undocumented fields do exist) 

and by the way UI is going to add some new fields like fieldId to each entity that has nothing to do with EG, but needed for UI. so its cannot be strict schema

Reply:
It makes perfectly sense - thanks a lot for the clarification 👍